From 19e1ae3af916ba0f7cdeadf12158389513ce062a Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sat, 17 Oct 2015 14:44:10 -0400
Subject: [PATCH 0611/1936] We no long need to transform valuators with
 negative minimums. Rounding in the integer widgets takes care of this.
 Transforms are needed only to provide a display in human terms.

---
 src/UI/OscilGenUI.fl | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/src/UI/OscilGenUI.fl b/src/UI/OscilGenUI.fl
index 6eb9eb85..14b3b780 100644
--- a/src/UI/OscilGenUI.fl
+++ b/src/UI/OscilGenUI.fl
@@ -346,9 +346,8 @@ oscildisplaygroup->redraw();}
             code1 {oscilo_base->parent(o);oscilo_base->init(true);}
           } {}
           Fl_Dial bfslider {
-            callback {redrawoscil();
-            bfparval->value(o->transform(o->value())) }
-            tooltip {Base Function Parameter} xywh {525 285 20 20} minimum 0 maximum 127 step 1
+            callback {redrawoscil(); bfparval->value(o->value());}
+            tooltip {Base Function Parameter} xywh {525 285 20 20} minimum -64 maximum 63 step 1
             code0 {o->init("Pbasefuncpar");}
             class Fl_Osc_Dial
           }
@@ -621,9 +620,8 @@ redrawoscil();}
             }
           }
           Fl_Dial wshpar {
-            callback {redrawoscil();
-            wsparval->value(o->transform(o->value())) }
-            tooltip {Waveshaping Parameter} xywh {265 318 20 20} minimum 0 maximum 127 step 1
+            callback {redrawoscil(); wsparval->value(o->value());}
+            tooltip {Waveshaping Parameter} xywh {265 318 20 20} minimum -64 maximum 63 step 1
             code0 {o->init("Pwaveshaping");}
             class Fl_Osc_Dial
           }
@@ -954,9 +952,6 @@ make_window();
 bftype->init("Pcurrentbasefunc");
 initialized = true;
 
-wshpar->set_transform(1,-64);
-bfslider->set_transform(1,-64);
-
 refresh();
 osceditUI->show();} {}
   }
-- 
2.47.0

